web_tree_date_search_extension/static/src/js/search_extension.js

```diff
@@ -0,0 +1,187 @@
+            return fn_mutex.exec(function () {
+                if (obj.isDestroyed()) { return $.when(); }
+                return fn.apply(obj, args)
+            });
+        };
+    }*/
+    // Replace the path
+    instance.web.views.map['list'] = 'instance.web_tree_date_search_extension.ListView';
+
+    instance.web_tree_date_search_extension.ListView = instance.web.ListView.extend({
+        init: function(parent, dataset, view_id, options) {
+            this._super.apply(this, arguments);
+
+            if ("dates_filter" in dataset.context){
+                this.dates_filter = dataset.context["dates_filter"]
+                this.current_date_from  = [];
+                this.current_date_to  = [];
+            }
+            this.search_extension_loaded = false;
+        },
+        do_load_state: function(state, warm) {
+            if (this.dates_filter && this.dates_filter.length > 0){
+                var $ui_toolbar_loc = $('.ui-toolbar:last').show();
+            }
+            else{
+                $('.ui-toolbar:last').hide();
+            }
+            return this._super.apply(this, arguments);
+        },
+        load_list: function(data) {
+            var self = this;
+            var tmp = this._super.apply(this, arguments);
+            if (!this.search_extension_loaded){
+                this.date_field_string = [];
+                for (i in this.dates_filter){
+                    var date_field = this.dates_filter[i];
+                    for (col in this.columns){
+                        if (this.columns[col].name == date_field){
+                            this.date_field_string[date_field] = this.columns[col].string;
+                            break;
+                        }
+                    }
+                }
+                if (this.dates_filter && this.dates_filter.length > 0){
+                    this.$el.parent().prepend(QWeb.render('SearchExtension', {widget: this}));
+                    var $ui_toolbar_loc = $('.ui-toolbar:last').show();
+                    //var $ui_toolbar_loc = $('.ui-toolbar:last').show().empty();
+                    for (i in this.dates_filter){
+                        var date_field = this.dates_filter[i];
+                        // add xml data
+                        var date_div = $("<div class='oe_form_dropdown_section' style='margin-right:8px; display: inline;'>" +
+                            "<h4 style='display: inline;'>" + this.date_field_string[date_field] + " :</h4>"+
+                            "<div class='oe_form' style='display: inline; margin-right:4px;' >"+
+                            "    <span class='oe_gantt_filter_from_" + date_field + "'></span>"+
+                            "</div>"+
+                            "<div class='oe_form' style='display: inline;' >"+
+                            "    <span class='oe_gantt_filter_to_" + date_field + "' ></span>"+
+                            "</div></div>");
```

Comment:
As I guess you said by ``add xml data`` it could be nice to use qweb template as it allow to overwrite it in sub module.